components/project-loader/project-loader.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import Service from '../../js/service.js';
+import ProjectList from '../project-list/project-list.jsx';
+
+export default React.createClass({
+  getInitialState() {
+    return {
+      loadingData: false,
+      apiPageIndex: 1,
+      entries: [],
+      moreEntriesToFetch: false,
+      totalMatched: 0,
+      tokenOfExistingPromise: null
+    };
+  },
+  getDefaultProps() {
+    return {
+      params: {}
+    };
+  },
+  componentDidMount() {
+    this.fetchData();
+  },
+  componentWillReceiveProps(nextProps) {
```

Comment:
I don't quite understand what you're doing in this function. Is there a reason we can't track this promise as a perfectly normal `this.promiseToken`? Why do we need to put it in state?